core/ir/opnd_shared.c

```diff
@@ -2795,6 +2797,30 @@ reg_get_size(reg_id_t reg)
     return OPSZ_NA;
 }
 
+#ifdef RISCV64
+/* Returns the OPSZ_ constant corresponding to the vector register size and lmul.
+ * Page 12 of RISC-V "V" Vector Extension Version 1.0.
+ */
+opnd_size_t
+reg_get_size_lmul(reg_id_t reg, lmul_t lmul)
+{
+    if (reg >= DR_REG_VR0 && reg <= DR_REG_VR31) {
+        int power = ((int)lmul << 29) << 29;
+        opnd_size_t opsz = opnd_size_from_bytes(dr_get_vector_length() >> (3 - power));
```

Comment:
So the 3 is to convert bits to bytes (worth a comment).  We then need to adjust for the LMUL: for our 8-at-a-time, we want to multiply by 3, so I assume you wanted `power` to be 3.  What about the 1/2, 1/4, 1/8?  Don't they need to shift right even more?  Is this `power` supposed to end up negative?